src/services/onchain-data.service.ts

```diff
@@ -249,6 +255,107 @@ export class OnchainDataService implements OnModuleInit
+            .filter([])
+            .range({
+                unit: "block",
+                // Since every block is 10s, 8640 blocks will be 24h
+                from: this.Connex.thor.status.head.number - 8640 * 60,
+                // Current block number
+                to: this.Connex.thor.status.head.number,
+            });
+
+        let end: boolean = false;
+        let offset: number = 0;
+        const limit: number = 256;
+
+        // Need a while loop because we can only get
+        // up to 256 events each round using connex
+        while (!end)
+        {
+            const result: Connex.Thor.Filter.Row<
+                "event", Connex.Thor.Account.WithDecoded
+            >[] = await swapFilter.apply(offset, limit);
+
+            for (const transaction of result)
+            {
+                // eslint-disable-next-line
+                if (!this.pointsPerAddress[transaction.decoded.sender])
+                {
+                    this.pointsPerAddress[transaction.decoded.sender] = {
+                        points: ethers.constants.Zero,
+                    };
+                }
+
+                this.pointsPerAddress[transaction.decoded.sender].points = this.pointsPerAddress[
+                    transaction.decoded.sender
+                ].points
+                    .add(transaction.decoded.amount0In)
+                    .add(transaction.decoded.amount0Out);
+            }
+
+            if (result.length === limit) offset += limit;
+            else end = true;
+        }
+    }
+
+    public getTradingCompetitionRanking(): IRankingItem[]
+    {
+        const pairInfo: IPair | undefined = this.getPair(this.rankingPairAddress);
+
+        if (!pairInfo) return [];
+
+        try
+        {
+            const rankingItems: IRankingItem[] = [];
+            Object.keys(this.pointsPerAddress).map((address: string) =>
+            {
+                rankingItems.push({
+                    address,
+                    points: this.pointsPerAddress[address].points,
+                    rank: 0,
+                });
+            });
+
+            // sort by the highest number of points
+            rankingItems.sort((a: IRankingItem, b: IRankingItem) =>
+            {
+                const pointsA: BigNumber = BigNumber.from(a.points);
+                const pointsB: BigNumber = BigNumber.from(b.points);
+
+                if (pointsA.gt(pointsB)) return -1;
+                if (pointsA.lt(pointsB)) return 1;
+                return 0;
+            });
+
+            // construct formatted payload
+            const ranking: IRankingItem[] = [];
+            rankingItems.map((item: IRankingItem, i: number) =>
+            {
+                ranking.push({
+                    address: item.address,
+                    points: formatEther(parseUnits(item.points.toString(), 18 - pairInfo.token0.decimals)),
+                    rank: i + 1,
+                });
+            });
```

Comment:
If we go ahead to use `Map` instead of object, we may not need this cuz `Map` preserves the order. Object doesn't preserve the order